modules/centreon-stream-connectors-lib/sc_metrics.lua

```diff
@@ -270,6 +294,13 @@ function ScMetrics:build_metric(format_metric)
   local metrics_info = self.metrics_info
 
   for metric, metric_data in pairs(self.metrics_info) do
+    if metrics_info[metric].instance ~= "" then
+      if #metrics_info[metric].subinstance ~= 0 then
+        metrics_info[metric].metric_name = metrics_info[metric].instance .. '~' .. table.concat(metrics_info[metric].subinstance, '~') .. '#' .. metrics_info[metric].metric_name
+      else
+        metrics_info[metric].metric_name = metrics_info[metric].instance .. '#' .. metrics_info[metric].metric_name
+      end
+    end
```

Comment:
```suggestion
    metrics_info[metric].metric_full_name = self:get_metric_full_name(metrics_info[metric])
```

we can't force the name of the metric to contain the whole instance and subinstances part. Depending on your situation, you may want to use them as tags of a metric and not as part of the name of the metric.
On the other hand, you may want an easy access to the full_metric_name.

if we want to grant access to the full metric name we need to add a new method for that